src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorFileChangeDetector.cs

```diff
@@ -72,17 +72,7 @@ public async Task StartAsync(string workspaceDirectory, CancellationToken cancel
             // Dive through existing Razor files and fabricate "added" events so listeners can accurately listen to state changes for them.
 
             workspaceDirectory = _filePathNormalizer.Normalize(workspaceDirectory);
-            
-            // Without trimming trailing `/` from the workspace directory, the FileSystemWatcher 
-            // returns with paths of the form   "workspaceDirectory/\\Pages\\Counter.razor"
-            // which are normalized to          "workspaceDirectory//Pages/Counter.razor" (Invalid `//`)
-            //
-            // This format doesn't match the directoryFilePaths we store as part of the Project Snapshot ->
-            //                                  "workspaceDirectory/Pages/Counter.razor"
-            // https://github.com/dotnet/aspnetcore-tooling/blob/488cf6e/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultRazorProjectService.cs#L328
-            // Consequently, files are being discarded into the MISC project and subsequently re-generated
-            workspaceDirectory = workspaceDirectory.TrimEnd('/');
```

Comment:
Okay, seems to be possibly tied to #1937 and https://github.com/dotnet/aspnetcore-tooling/pull/1943#issuecomment-635645650. 

Those have nothing to do with this PR / this fix. I've reverted #1937 here, and will merge in this PR.

Separately, I'll create a PR to address flakiness and re-introduce the fix for #1937.